src/main/main.cpp

```diff
@@ -340,28 +342,43 @@ catchup(Config const& cfg, uint32_t to, uint32_t count)
 }
 
 static int
-catchupAt(Config const& cfg, uint32_t at)
+catchupAt(Config const& cfg, uint32_t at, Json::Value& catchupInfo)
 {
-    return catchup(cfg, at, 0);
+    return catchup(cfg, at, 0, catchupInfo);
 }
 
 static int
-catchupComplete(Config const& cfg)
+catchupComplete(Config const& cfg, Json::Value& catchupInfo)
 {
     return catchup(cfg, CatchupConfiguration::CURRENT,
-                   std::numeric_limits<uint32_t>::max());
+                   std::numeric_limits<uint32_t>::max(), catchupInfo);
 }
 
 static int
-catchupRecent(Config const& cfg, uint32_t count)
+catchupRecent(Config const& cfg, uint32_t count, Json::Value& catchupInfo)
 {
-    return catchup(cfg, CatchupConfiguration::CURRENT, count);
+    return catchup(cfg, CatchupConfiguration::CURRENT, count, catchupInfo);
 }
 
 static int
-catchupTo(Config const& cfg, uint32_t to)
+catchupTo(Config const& cfg, uint32_t to, Json::Value& catchupInfo)
 {
-    return catchup(cfg, to, std::numeric_limits<uint32_t>::max());
+    return catchup(cfg, to, std::numeric_limits<uint32_t>::max(), catchupInfo);
+}
+
+static void
+writeCatchupInfo(Json::Value const& catchupInfo, std::string const& outputFile)
+{
+    std::string filename = outputFile.empty() ? "catchupInfo.json" : outputFile;
```

Comment:
I don't know if this makes sense: I'd rather have a behavior consistent across all command lines that use "--output-file".
Something in the lines of a common function that takes a string and where the default filename is "-" (ie stdout) - so if people want to output to a file, they have to specify a file name. I know that breaks the current contract for "graphquorum" but it seems worth it instead of having different implementations and behaviors for different command line options.

Reply:
Behavior of `--output-file` is consistent:

writeCatchupInfo:
```
std::string filename = outputFile.empty() ? "catchupInfo.json" : outputFile;
```

reportLastHistoryCheckpoint:
```
std::string filename =
            outputFile.empty() ? "lasthistorycheckpoint.json" : outputFile;
```

writeQuorumGraph:
```
std::string filename = outputFile.empty() ? "quorumgraph.dot" : outputFile;
```

Currently we cannot support `-` as stdout, as all of our logging information goes to stdout instead of stderr (I've covered that in #1418). Should I do #1418 first so we can implement `-`?

Reply:
what I am saying is that the behavior is not consistent as the default filename depends on the command (so you can't output the default filename in the help for `--output-file` and you can't refactor the `output-file` functionality into its own function).

I don't think there is a dependency on #1418 for this:
what I am suggesting is that when people run from a shell, outputting the result to stdout allows people to see what the command did and if they want to use the output for something else, they can just use the `--output-file` option to a temp file for example.

Cleaning up stdout vs stderr will allow people to pipe commands (and not use a temp file), which is a different thing.

Reply:
So what should I do here? Implement #1427 and add support for `-`? Change default name of file for output to output.json for all commands? Or add LOG output for all of them?